src/ostree/ot-builtin-checkout.c

```diff
@@ -136,7 +136,15 @@ process_one_checkout (OstreeRepo           *repo,
       else if (opt_union_add)
         options.overwrite_mode = OSTREE_REPO_CHECKOUT_OVERWRITE_ADD_FILES;
       else if (opt_disjoint_union)
-        options.overwrite_mode = OSTREE_REPO_CHECKOUT_OVERWRITE_DISJOINT_UNION_FILES;
+        {
+          if (!opt_require_hardlinks)
```

Comment:
If we didn't require it, we'd have to change `create_file_copy_from_input_at()` to also handle the merge (right now we abort), and it'd be harder since we'd actually need to checksum the files rather than comparing by device+inode.  Since I don't know of a good use case for that, I'd prefer not supporting it and only do so if we have to.

Reply:
ah I see, it makes sense. Thanks for the explanation :)